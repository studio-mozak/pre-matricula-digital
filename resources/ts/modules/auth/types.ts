export interface Auth {
  level: number;
  name: string;
  schools: string[];
}
