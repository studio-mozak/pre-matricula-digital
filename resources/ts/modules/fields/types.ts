import { Nullable } from '@/types';

export type SetValues = (fields: Record<string, unknown>) => void;

export interface FieldPageOption {
  id?: string;
  name: Nullable<string>;
  weight: number;
}

export interface FieldPage {
  id: Nullable<string>;
  name: Nullable<string>;
  description: Nullable<string>;
  group: Nullable<'RESPONSIBLE' | 'STUDENT'>;
  type: Nullable<string>;
  internal: boolean;
  options: FieldPageOption[];
  deleteOptions?: string[];
  required?: boolean;
}

export interface ErrorResponse {
  message: string;
}
