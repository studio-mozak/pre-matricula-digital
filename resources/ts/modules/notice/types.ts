import { Nullable } from '@/types';

export interface Notice {
  id: Nullable<number>;
  text: Nullable<string>;
}

export interface NoticeSubmitResponse {
  errors: boolean;
  id?: number;
}
