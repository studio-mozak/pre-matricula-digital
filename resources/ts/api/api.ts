import axios, { AxiosInstance } from 'axios';

const token = window.config ? window.config.token : '';

export const http: AxiosInstance = axios.create({
  baseURL: String(import.meta.env.VITE_APP_API_URL || ''),
  headers: {
    Authorization: `Token ${token}`,
  },
});

axios.interceptors.response.use(
  (res) => {
    /**
     * If the response from the server has the "errors" object or the object "data" is empty,
     * here it is forced to go to the catch block automatically.
     */
    if (
      (res.data.errors || Object.keys(res.data).length === 0) &&
      !(res.data instanceof Blob)
    ) {
      return Promise.reject({
        response: {
          data: res.data,
        },
      });
    }

    return res;
  },
  (err) => Promise.reject(err)
);

export const rest = <T>(
  type: 'get' | 'post' | 'put' | 'delete',
  url: string,
  params?: object
): Promise<T> => http[type](url, params);

export const graphql = <T>(data: object, config?: object): Promise<T> =>
  axios.post('/graphql', data, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
    ...config,
  });
