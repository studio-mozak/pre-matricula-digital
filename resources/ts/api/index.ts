export * from './api';
export * as City from './services/graphql/city';
export * as Resource from './services/graphql/resource';
