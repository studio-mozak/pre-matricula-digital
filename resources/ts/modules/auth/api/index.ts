/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as AuthRest from '@/modules/auth/api/services/rest/auth';
