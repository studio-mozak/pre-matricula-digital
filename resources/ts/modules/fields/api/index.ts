/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Field from './services/graphql/field';
