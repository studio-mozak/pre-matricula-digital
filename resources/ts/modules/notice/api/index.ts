/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Notice from './services/graphql/notice';
