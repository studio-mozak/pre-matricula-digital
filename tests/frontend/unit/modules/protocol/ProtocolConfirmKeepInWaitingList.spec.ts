import { ProtocolStatusPreRegistration } from '@/modules/protocol/types';
import { Protocol as ProtocolApi } from '@/modules/protocol/api';
import { flushPromises, mount, VueWrapper } from '@vue/test-utils';
import { expect, vi } from 'vitest';
import VTooltip from 'v-tooltip';
import ProtocolConfirmKeepInWaitingList from '@/modules/protocol/components/ProtocolConfirmKeepInWaitingList.vue';

vi.mock('@/modules/protocol/api');

const preregistrationMock: ProtocolStatusPreRegistration = {
  classroom: {
    grade: {
      name: 'string',
    },
    name: 'string',
    period: {
      name: 'string',
    },
  },
  id: 'string',
  observation: 'string',
  parent: null,
  position: 1,
  process: {
    showPriorityProtocol: false,
    blockIncompatibleAgeGroup: false,
    forceSuggestedGrade: false,
    grades: [],
  },
  school: {
    area_code: 1,
    name: 'string',
    phone: 'string',
  },
  stage: {
    observation: 'string',
  },
  status: 'IN_CONFIRMATION',
  student: {
    dateOfBirth: 'string',
    initials: 'string',
  },
  type: 'string',
  waiting: null,
};

const onMounted = async (preregistration: {
  modelValue: boolean;
  preregistration: ProtocolStatusPreRegistration;
}) => {
  const wrapper = mount(ProtocolConfirmKeepInWaitingList, {
    props: preregistration,
    global: {
      stubs: ['modal'],
      directives: {
        tooltip: VTooltip,
      },
    },
  });

  await wrapper.vm.$nextTick();

  await flushPromises();

  return wrapper;
};

describe('ProtocolConfirmKeepInWaitingList', () => {
  let wrapper: VueWrapper<
    InstanceType<typeof ProtocolConfirmKeepInWaitingList>
  >;

  beforeEach(async () => {
    wrapper = await onMounted({
      modelValue: true,
      preregistration: preregistrationMock,
    });

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('returnToWait must be executed successfully', async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockResolvedValueOnce({
      errors: false,
      keepOnTheWaitingList: {
        id: '1',
      },
    });

    const dialogSpy = vi.spyOn(wrapper.vm.ctx, 'dialog');

    await wrapper.vm.$nextTick();

    wrapper.vm.returnToWait();

    await flushPromises();

    expect(dialogSpy).toBeCalledTimes(1);
    expect(dialogSpy).toBeCalledWith({
      title: 'Sucesso!',
      description:
        'A permanência na lista de espera foi confirmada com sucesso.',
    });
  });

  test('returnToWait must be executed with errors', async () => {
    vi.mocked(ProtocolApi.postReturnToWait).mockRejectedValueOnce({
      response: {
        data: {
          errors: true,
        },
      },
    });

    const dialogSpy = vi.spyOn(wrapper.vm.ctx, 'dialog');

    await wrapper.vm.$nextTick();

    wrapper.vm.returnToWait();

    await flushPromises();

    expect(dialogSpy).toBeCalledTimes(1);
    expect(dialogSpy).toBeCalledWith({
      title: 'Erro!',
      description:
        'Não foi possível confirmar a permanência na lista de espera. ' +
        'Por favor, entre em contato com o suporte.',
      titleClass: 'danger',
    });
  });
});
