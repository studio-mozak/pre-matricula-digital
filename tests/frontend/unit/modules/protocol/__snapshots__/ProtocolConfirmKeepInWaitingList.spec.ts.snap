// Vitest Snapshot v1

exports[`ProtocolConfirmKeepInWaitingList > should match snapshot 1`] = `
<modal-stub
  bodyclass=""
  fullwidth="false"
  iconleft=""
  iconright=""
  initialloading="false"
  isoverflowvisible="false"
  large="false"
  loadinginputs="false"
  loadingtext="false"
  maximized="false"
  modelvalue="true"
  nofooter="false"
  noheader="false"
  overflowvisible="false"
  persistent="false"
  position="standard"
  seamless="false"
  title="Deseja continuar na lista?"
  titleclass="font-hind-18-primary text-primary"
  tooltipicon=""
/>
`;
